modules/integration/tests-integration/bpmn/src/test/java/org/wso2/bps/integration/tests/bpmn/BPMNMultiTenancyTestCase.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.bps.integration.tests.bpmn;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.bps.integration.common.clients.bpmn.WorkflowServiceClient;
+import org.wso2.bps.integration.common.utils.BPSMasterTest;
+import org.wso2.carbon.bpmn.core.mgt.model.xsd.BPMNDeployment;
+
+import java.util.concurrent.TimeUnit;
+
+public class BPMNMultiTenancyTestCase extends BPSMasterTest {
+    private static final Log log = LogFactory.getLog(BPMNTaskCreationTestCase.class);
+
+    private WorkflowServiceClient workflowServiceClient;
+    private int deploymentCount;
+    boolean accessedSameArtifactInstance = false;
+    String domainKey1 = "wso2.com";
+    String userKey1 = "user1";
+    String domainKey2 = "abc.com";
+    String userKey2 = "user2";
+
+
+    @BeforeClass(alwaysRun = true)
+    public void createTenant() throws Exception {
+
+        // initialize for tenant wso2.com
+        initialize(domainKey1, userKey1);
+
+    }
+
+    //initialize for each tenant
```

Comment:
Thanks for pointing it out, Improved with method comments. Updated fix in commit b8002ea